src/ledger/utils.ts

```diff
@@ -29,26 +30,24 @@ function getXRPBalance(
     account: address,
     ledger_index: ledgerVersion
   }
-  return connection
+  const data = await connection
     .request(request)
-    .then((data) => common.dropsToXrp(data.result.account_data.Balance))
+  return common.dropsToXrp(data.result.account_data.Balance)
 }
 
 // If the marker is omitted from a response, you have reached the end
-function getRecursiveRecur(
+async function getRecursiveRecur(
```

Comment:
What is this function supposed to do? (I'm not quite clear on when I'd use it from the name)

Reply:
I'm not really sure because I don't understand the getter/setter paradigm, but messing with this is out of the scope of this PR.